blockly/frontend/src/generators/java/arrays.ts

```diff
@@ -46,7 +46,7 @@ export function array_get(
 
 export function array_length(
   block: Blockly.Block,
-  generator: Blockly.Generator
+  _generator: Blockly.Generator
```

Comment:
Weshalb der underscore?

Reply:
Die Methode verwendet den Parameter nicht, weshalb sich der Compiler beschwert. Nicht verwendete Parameter, Felder, Rückgabewerte, ... werden mit einem Unterstrich gekennzeichnet. Ich wollte erstmal die Logik nicht ändern, ich wollte nur schnell die Style-Check Fehler beheben, ich weiß nicht, ob mit dem Parameter etwas anderes geplant war.

Reply:
Danke. Der "_" kann in TypeScript ein Zeichen für einen Member sein. Aber das verwenden eigentlich nur die Entwickler, nicht die Community - und ist eigentlich out of date.

Wie dem auch sei, es obliegt, imho, dem Code-Owner, hierbei den Style festzulegen. Siehe auch [hier](https://stackoverflow.com/questions/47747809/typescript-underscore-convention-for-members).

Reply:
Kannst du mir die Fehlermeldung einmal zeigen? Ich erhalte da nämlich keine.

Es geht darum, dass die Variablennamen zurzeit nicht einheitlich sind:

```
$ grep -B 3 generator: blockly/frontend/src/generators/java/arrays.ts

export function var_array(
  block: Blockly.Block,
  _generator: Blockly.Generator
--

export function array_set(
  block: Blockly.Block,
  generator: Blockly.Generator
--

export function array_get(
  block: Blockly.Block,
  generator: Blockly.Generator
--

export function array_length(
  block: Blockly.Block,
  _generator: Blockly.Generator
```

Reply:
```cmd
src/generators/java/arrays.ts:6:3 - error TS6133: 'generator' is declared but its value is never read.

6   generator: Blockly.Generator
    ~~~~~~~~~


Found 1 error in src/generators/java/arrays.ts:6
```

Da in der `tsconfig.json`:
```
"noUnusedParameters": true,
```
eingestellt ist.

Aber wie gesagt _ ist für nicht verwendete Parameter, in fast jeder Sprache.

Reply:
Habe npm zurzeit nicht installiert. Aber nach dieser Diskussion: https://github.com/Microsoft/TypeScript/issues/9458 sollten wir die Option "noUnusedParameters" entweder deaktivieren, oder alle Parameternamen sollten mit _ anfangen.

Reply:
ich finde die diskussion grad nicht zielführend. es geht in diesem pr sus meiner sicht nicht um coding-style oder die änderung von coding conventions. außerdem kostet alleine diese diskussion zeit, die wir im projekt nicht haben.

@tgrothe wenn du die diskussion um namensgebung im typescript wirklich wichtig findest, mache bitte ein neues ticket auf, beschreibe das problem nachvollziehbar und stelle ein oder zwei Lösungsvorschlägen vor. dann können wir das separat diskutieren.

Reply:
Danke, das Ticket ist auf.